### ISSUE TITLE
UNC user-data-dir means launch fails silently

### ISSUE THREAD

**Issue by cbull-86** [microsoft/vscode#182073]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.78.1
- OS Version: Windows 10

Steps to Reproduce:

1. Install the latest version and try and launch the application with a UNC path as the user-data-dir
4 processes will spawn but with no user interface. There is zero user feedback suggesting why the application has not launched successfully.

**Comment by hdeadman** [comment]
```
C:\Program Files\Microsoft VS Code>code

C:\Program Files\Microsoft VS Code>
Error [ERR_UNC_HOST_NOT_ALLOWED]: UNC host 'ndevfs' access is not allowed
    at Object.readFile (node:fs:402:10)
    at t.readFile (node:electron/js2c/asar_bundle:2:8867)
    at i._readSourceAndCachedData (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:5:4004)
    at i.load (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:5:1930)
    at d.load (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:4:12351)
    at c (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:6:9301)
    at h._loadModule (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:6:9429)
    at h._resolve (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:7:452)
    at h.defineModule (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:6:5524)
    at i (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:7:2149)
    at _.load (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:7:7717)
    at startup (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:7:9977)
    at onReady (C:\Program Files\Microsoft VS Code\resources\app\out\main.js:7:10204) {
  code: 'ERR_UNC_HOST_NOT_ALLOWED'
}
```
I was able to launch by setting APPDATA to a local folder but my settings are on the roaming folder.

**Comment by cbull-86** [comment]
@hdeadman Yes this was my eventual solution but only after several hours of fruitless troubleshooting...

**Comment by hdeadman** [comment]
This was reported here, apparently it is on purpose: https://github.com/microsoft/vscode/issues/182070

**Comment by cbull-86** [comment]
I saw that before posting - I raised the issue as I don't feel a silent failure to be particularly user friendly - as I said I wasted hours troubleshooting the issue before coming across the message about working with UNC paths in the docs. I thought it more likely to be an environment issue as there was no warning message. (I'm aware that if you launch from the command line you will see one but I was not).

**Comment by bpasero** [comment]
For insiders, we will automatically allow the UNC path if it is configured as user data dir: https://github.com/microsoft/vscode/issues/182024. You can already try that out from: https://code.visualstudio.com/insiders/

For stable: we have a fix in works for `1.78.2` that you can set an environment variable `NODE_UNC_HOST_ALLOWLIST` before launching VS Code. That unfortunately does not work in `1.78.1`.

**Comment by kragil** [comment]
I still have this problem with 1.78.2.
